src/app/donation-complete/donation-complete.component.html

```diff
@@ -14,115 +17,97 @@
-          <p class="b-rt-1 b-m-0" *ngIf="campaign.parentRef"><a [routerLink]="'/' + campaign.parentRef">Explore other campaigns</a></p>
-          <p class="b-rt-1 b-m-0" *ngIf="!campaign.parentRef"><a routerLink="/explore">Explore other campaigns</a></p>
-        </div>
-        <a routerLink="/"><img class="c-button__image" src="/assets/images/logo.png" alt="The Big Give" width="200"></a>
-      </button>
+      <p *ngIf="registrationComplete" id="registrationCompleteText">
+        <strong>You are now registered<span *ngIf="loggedIn"> and logged in</span>.</strong> You can now donate faster using your chosen email address and password.
+      </p>
+
+      <p *ngIf="registerError" class="error" aria-live="assertive">
+        Your password could not be set. Please ensure you chose one with at least 10 characters.
+      </p>
+    </div>
+
+    <hr>
+
+    <div class="donation-summary">
+      <p><strong>Hi Ali,<br/>Your donation summary</strong></p>
+      <p>Your donation of <strong class="figure">{{ donation.donationAmount | exactCurrency:donation.currencyCode }}</strong> has been made. That's worth <strong class="figure">{{ totalValue | exactCurrency:donation.currencyCode }}</strong> to {{ donation.charityName }}!</p>
+
+      <br/>
+      <p *ngIf="campaign?.thankYouMessage">{{ campaign.thankYouMessage }}</p>
+    </div>
+
+    <hr>
+
+    <div class="receipt">
+      <p><strong>Your Receipt</strong></p>
+      <p>{{donation.createdTime | date: 'medium'}}</p>
+      <br>
+
+      <table>
+        <tr>
+          <td>Your donation</td>
+          <td>{{ donation.donationAmount | exactCurrency:donation.currencyCode }}</td>
+        </tr>
+        <tr *ngIf="donation.matchedAmount > 0">
+          <td>Matched Amount</td>
+          <td>{{ donation.matchedAmount | exactCurrency:donation.currencyCode }}</td>
+        </tr>
+        <tr *ngIf="giftAidAmount > 0">
+          <td>Gift Aid Amount</td>
+          <td>{{ giftAidAmount | exactCurrency:donation.currencyCode }}</td>
+        </tr>
+        <tr *ngIf="donation.tipAmount > 0">
+          <td>Tip Amount</td>
+          <td>{{ donation.tipAmount | exactCurrency:donation.currencyCode }}</td>
+        </tr>
+        <tr class="receives">
+          <td>Charity Receives</td>
+          <td>{{ totalValue | exactCurrency:donation.currencyCode }}</td>
+        </tr>
+        <tr class="total">
+          <td><strong>Total amount you have been charged</strong></td>
+          <td><strong>{{ cardChargedAmount | exactCurrency:donation.currencyCode }}</strong></td>
+        </tr>
+      </table>
+    </div>
+
+    <hr>
+
+    <div class="socials">
+      <p>Boost your impact by sharing on social media</p>
+      <div class="social-icons-wrapper">
+        <biggive-social-icon service="Facebook" url="https://www.facebook.com/sharer/sharer.php?u={{ shareUrl }}" background-colour="tertiary" icon-colour="black" wide="true"></biggive-social-icon>
```

Comment:
Does the shareUrl not need to be encoded with encodeURIComponent before being used as a compnent of a URI here and in the next lines?

(btw I looked up [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) and [encodeURI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI), I don't understand what the point of the latter function existing is)

Reply:
I see that `prefilledText` is already encoded with encodeURIComponent in the .ts file. Might be worth either renaming that variable to make it clear that it's encoded, or moving the call to encode to the template file.

Reply:
Fixed.